containers/daap-resync-unprocessed-files/src/var/task/resync_unprocessed_files.py

```diff
@@ -16,77 +24,62 @@
         data_product_prefix=raw_prefix,
     )
 
-    # Find extraction timestamps in the raw area, and not in the curated area
-    paginator = s3.get_paginator("list_objects_v2")
-    curated_pages = paginator.paginate(
-        Bucket=curated_data_bucket, Prefix=curated_prefix
+    # get data product has associated curated data
+    curated_pages = get_data_product_pages(
+        bucket=curated_data_bucket, data_product_prefix=curated_prefix
     )
 
-    # key = "raw_data/data_product/table/extraction_timestamp=timestamp/file.csv"
-    raw_table_timestamps = get_raw_data_unique_extraction_timestamps(raw_pages)
-
-    curated_table_timestamps = set()
-    for item in curated_pages.search("Contents"):
-        # key = "curated_data/database_name=data_product/table_name=table"
-        #       + "/extraction_timestamp=timestamp/file.parquet"
-        if item["Size"] > 0:
-            data_product = search_string_for_regex(
-                string=item["Key"], regex=database_name_regex()
-            )
-
-            table = search_string_for_regex(
-                string=item["Key"], regex=table_name_regex()
-            )
-
-            extraction_timestamp = search_string_for_regex(
-                string=item["Key"], regex=extraction_timestamp_regex()
-            )
-
-            # Both sets need the same formatting to compare them
-            curated_table_timestamps.add(
-                f"{data_product}/{table}/{extraction_timestamp}"
-            )
+    raw_table_timestamps = get_unique_extraction_timestamps(raw_pages)
+    curated_table_timestamps = get_curated_unique_extraction_timestamps(curated_pages)
 
-    timestamps_to_resync = raw_table_timestamps - curated_table_timestamps
-    raw_keys_to_resync = [
-        item["Key"]
-        for timestamp in timestamps_to_resync
-        for item in raw_pages.search("Contents")
-        if timestamp in item["Key"]
-    ]
+    # compare and filter the raw files to sync
+    raw_keys_to_resync = get_resync_keys(
+        raw_table_timestamps, curated_table_timestamps, raw_pages
+    )
     logger.info(f"raw keys to resync: {raw_keys_to_resync}")
 
     # Feed unprocessed data files through the load process again.
     # If there are over 1000 files, the lambda will get jammed up
     aws_lambda = boto3.client("lambda")
     for key in raw_keys_to_resync:
-        payload = f'{{"detail":{{"bucket":{{"name":"{raw_data_bucket}"}}, "object":{{"key":"{key}"}}}}}}'
+        payload = f'{{"detail":{{"bucket":{{"name":"{raw_data_bucket}"}},\
+        "object":{{"key":"{key}"}}}}}}'
         aws_lambda.invoke(
             FunctionName=athena_load_lambda, InvocationType="Event", Payload=payload
         )
 
     logger.info(
-        f"data product {data_product_to_recreate} resynced with {len(raw_keys_to_resync)} files"
+        f"data product {data_product_to_recreate} resynced"
+        + "with {len(raw_keys_to_resync)} files"
     )
     logger.info(str(raw_keys_to_resync))
```

Comment:
Is this required ? Keys are logged on line 68.

Reply:
Will be useful for debugging